packages/block-library/src/post-navigation-link/index.php

```diff
@@ -116,13 +112,98 @@ function render_block_core_post_navigation_link( $attributes, $content ) {
 		$content = $get_link_function( $format, $link );
 	}
 
+	$support_styles = block_core_post_navigation_link_get_border_and_spacing_attributes( $attributes );
+	$classes       .= ! empty( $support_styles['class'] ) ? " {$support_styles['class']}" : '';
```

Comment:
Similar to the editor side flagged above, this change doesn't take into account that the border support can apply color CSS classes e.g. `has-border-color`, `has-accent-1-border-color` etc for preset colors for "flat" borders.

The result is the issue raised in my last review is still present.